R/FilterStateChoices.R

```diff
@@ -15,6 +16,66 @@
 #' isolate(filter_state$set_selected("B"))
 #' isolate(filter_state$set_keep_na(TRUE))
 #' isolate(filter_state$get_call())
+#'
+#' \dontrun{
+#' # working filter in an app
+#' library(shiny)
+#'
+#' data_choices <- c(sample(letters[1:4], 100, replace = TRUE), NA)
+#' filter_state_choices <- ChoicesFilterState$new(
+#'   x = data_choices,
+#'   varname = "variable",
+#'   varlabel = "label"
+#' )
+#' filter_state_choices$set_state(list(selected = data_choices[c(1, 3)], keep_na = TRUE))
```

Comment:
`data_choices[1]` and `data_choices[3]` can be identical - I don't know what can happen if one sets `selected = c("a", "a")`. Also `data_choices[1]` or `data_choices[3]` can be `NA` which is wrong

```suggestion
#' filter_state_choices$set_state(list(selected = data_choices[c(1, 3)], keep_na = TRUE))
```

Reply:
Indeed, fixed.